src/main/java/com/mraof/minestuck/entity/consort/MessageType.java

```diff
@@ -1036,39 +1038,7 @@ public MutableComponent getMessage(ConsortEntity consort, ServerPlayer player, S
 				return conditionedMessage.getMessage(consort, player,
 						addTo(chainIdentifier, conditionedMessage.getString()));
 			
-			boolean hasItem = false;
-			if(random || repeat && nbt.contains(this.getString()))
-			{
-				int index;
-				if(nbt.contains(this.getString()))
-					index = nbt.getInt(this.getString());
-				else
-				{
-					index = consort.level.random.nextInt(possibleItems.size());
-					nbt.putInt(this.getString(), index);
-				}
-				
-				ItemStack stack = possibleItems.get(index);
-				nbt.put(this.getString() + ".item", stack.save(new CompoundTag()));
-				
-				hasItem = lookFor(stack, player);
-			} else
-			{
-				List<ItemStack> list = new ArrayList<>(possibleItems);
-				while (!list.isEmpty())
-				{
-					ItemStack stack = list.remove(consort.level.random.nextInt(list.size()));
-					if(lookFor(stack, player))
-					{
-						nbt.putInt(this.getString(), possibleItems.indexOf(stack));
```

Comment:
there is no longer an index to utilize for this nbt so it has been removed. Overall let me know if there is anything I cut out from the original code that you would like to be returned

Reply:
The index was originally used to identify which of the possible items in the list that the message type has targeted. As far as I can tell, you could use the resource name to identify which item in the tag that is targeted.

Also, some of the functionality seems to have been lost. I'd suggest checking out the javadoc on the constructor if you haven't already.

Reply:
Switched the index to instead be a string of the registry name. Looking at nbt and playing out a consort interaction, the functionality appears to be restored. The text does not repeat and the consort only asks for a specific item if it was being held.

`MessageData: {380df991-f603-344c-a090-369bad2a924a: {hungry.item: {id: "minecraft:cookie", Count: 1b}, hungry: 0, currentMessage: "hungry.ask_food"}}`